Proofs/Popcount32.lean

```diff
@@ -71,6 +71,7 @@ def popcount32_program : Program :=
 
 #genStepEqTheorems popcount32_program
 
+set_option maxHeartbeats 0 in
```

Comment:
The need to do this for popcount worries me. What's going on here?

Reply:
@shigoel cleaned up, was `simp_mem` being slow by whacking `omega` at everything. I think this is a mistake, so I'll send a follow up PR that reduces the aggressiveness of `simp_mem`.